app/Judite/Models/Student.php

```diff
@@ -159,4 +160,85 @@ public function scopeWhereNumber($query, $studentNumber)
     {
         return $query->where('student_number', $studentNumber);
     }
+
+    /**
+     * Get memberships of this student.
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function memberships()
+    {
+        return $this->hasMany(Membership::class);
+    }
+
+    /**
+     * Make this student member of a given group.
+     *
+     * @param \App\Judite\Models\Group $group
+     *
+     * @throws \App\Exceptions\UserHasAlreadyGroupInCourseException
+     *
+     * @return \App\Judite\Models\Group
+     */
+    public function join(Group $group): Membership
+    {
+        if ($this->isMemberOfGroupInCourse($group->course_id)) {
+            throw new UserHasAlreadyGroupInCourseException();
+        }
+
+        $membership = $this->memberships()->make();
```

Comment:
I'm not sure of this, but let know what do you think.
```
$this->memberships()->create([
    'group_id' => $group->id,
    'course_id' => $group->course_id,
]);
```

Reply:
Used the same logic as the enroll method and like this approach better.